src/mixins/object_ancestry.mixin.js

```diff
@@ -1,10 +1,31 @@
 fabric.util.object.extend(fabric.Object.prototype, /** @lends fabric.Object.prototype */ {
 
   /**
-     *
-     * @param {boolean} [strict] returns only ancestors that are objects (without canvas)
-     * @returns {(fabric.Object | fabric.StaticCanvas)[]} ancestors from bottom to top
-     */
+   * Checks if object is decendant of target
+   * Should be used instead of @link {fabric.Collection.contains} for performance reasons
+   * @param {fabric.Object|fabric.StaticCanvas} target
+   * @returns {boolean}
+   */
+  isDescendantOf: function (target) {
+    var parent = this.group || this.canvas;
+    while (parent) {
+      if (target === parent) {
+        return true;
+      }
+      else if (parent instanceof fabric.StaticCanvas) {
+        //  happens after all parents were traversed through without a match
+        return false;
+      }
+      parent = parent.group || parent.canvas;
+    }
+    return false;
+  },
+
+  /**
+   *
+   * @param {boolean} [strict] returns only ancestors that are objects (without canvas)
```

Comment:
i wonder how we will find ourselves with the following:
typescript and getAncestor returning an Object or a Canvas
the other ancestry methods not having the strict option.

Reply:
checkout [typescript playground](https://www.typescriptlang.org/play?#code/FAQwRgzgLgTiDGUAE8A2IISQMXDAlvAPJgBWApokgN7BJLQhSFIzkgAmA9gHaoCeSKPwAO5AFwNY+HgHM6DclADCXLjA4QAFAEoawBQF9gx4GgxYASpWTkAHlHI9NOPIRIUqteo2bxW7Nx8gsJikgDkbIjhJgbmmEjK+DBo5Ej2js5YuGAExGQ2+j5QTCxsnLwCQqISSOHwyakxpvFYAApMABbpDk4uOXkehd5Spf7lQVWhteEiXc1x6AnKIDwAbhhFAc7kMACCqKi6kmtc+Bz6RrFmS1hDiHs88OTQMIIjskqPz9DqEAA8ABUepkXGA1Kh2DwAHxaV6EKAAfkkgJ0KJBfSwsAArmlEa5cu4CogANoAXSQkhJADpaQMiZ4oOSADSJVYbCAUkb0eC8aBIEBIAC8SB45AA7khrIhdABuBQ8vnIMDC0USxKNSFyhUoJUoVViyUdKCdbX0RU8fnwdmbEWGtnrDBm81sKDYmA8KR5ZD4kkgZlgZnwClU-2B+BBm2c2VIAD0sdFXCQ53YSHFnSY4Sw1uxEBksiEnXwWF2MHUV3onyg3xeUD+ACYgRiskhwVxIatYfDEMikKj0RlMUIYLikPj6flGeTKUgaXS3JObCyHRyuTreZbkIK7erpVBnRb+Sqd5KkiktTp5ebdZv9SekMbTZf13rrY6sPeVu+D3HYwABKAIAAWnwWQeHUcgdVdd1PW7H1ZzDIMQwQgMg0jd8ySvJBjFMe17mrJ5azeXRqSrGtfhgbQdGpEQuBEXRfwJQZiWQAAfFdNnY7EdgAMxkcgOGAPCWPI2B+BIsjCIo7QcXIajaPovR4yYhlCi43j+MEoT1Xw0TiOoySfjrSitFk+S6N0RFqQgJRVHUTQ5SAA)

Reply:
hover on top of the results to see the types